recipe/meta.yaml

```
@@ -45,36 +47,46 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    - {{ compiler('cuda') }}  # [cuda_compiler_version != "12.0"]
+    - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    - sysroot_{{ target_platform }} 2.17  # [linux]
   host:
+    - cuda-version {{ cuda_major }}.0
     - patchelf >=0.12  # [linux]
     # to suppress ".so not found" errors
-    - libzlib  # [linux]
+    - libzlib       # [linux]
     - libzlib-wapi  # [win64]
   run:
+    - {{ pin_compatible("cuda-version", max_pin="x") }}
+    {% if cuda_major == "11" %}
     - cudatoolkit 11.*
+    {% else %}
+    - cuda-cudart-dev
+    - cuda-nvrtc
+    - libcublas
+    {% endif %}
   run_constrained:
-    # Only GLIBC_2.17 or older symbols present
-    - __glibc >=2.17  # [linux]
+    - __glibc >=2.17                        # [linux and not (aarch64 and cuda_major == "12")]
+    - __glibc >=2.28                        # [linux and (aarch64 and cuda_major == "12")]
```

I think this should be just
```suggestion
    - __glibc >=2.17                        # [linux and not aarch64]
    - __glibc >=2.28                        # [linux and aarch64]
```
(I think `linux` can be omitted too). I see the glibc 2.27 symbols in CUDA 11 binaries as well.

---

We need the linux constraint since this also has Windows builds.

---

Right but `aarch64` evaluates to False on `win-32`/`win-64`.

---

btw to keep for record, this is what I see with the CUDA 11 + aarch64 tarball:
```
$ for i in *.so; do echo $i; readelf -s $i | grep GLIBC | grep -v 'GLIBC_2.17'; done
libcudnn.so
libcudnn_adv_infer.so
libcudnn_adv_train.so
libcudnn_cnn_infer.so
   331: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND log2f@GLIBC_2.27 (12)
libcudnn_cnn_train.so
   118: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND powf@GLIBC_2.27 (8)
   250: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND log2f@GLIBC_2.27 (8)
   551: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND expf@GLIBC_2.27 (8)
libcudnn_ops_infer.so
libcudnn_ops_train.so
```